### PR TITLE
ipc_service: Make the backend sample more realistic

### PR DESCRIPTION
**Pull request by carlocaione**
The current static_vrings backend sample is allocating all host
instances on one core and all remote instances on the other core.

Change this by mixing up the roles of the instances to make the sample
more useful, complex and more close to the actual real-life usage.

Change the sample to dynamically print the role instead of hardcoding
it.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

### CODE REVIEW

**Comment by arnopo** [comment]
Could you detail what you mean by "more close to the actual real-life usage"?
The samples using OpenAMP have been designed to optimize the code for the remote side (i.e. the coprocessor).
For this the main processor is usually set to host role and and the remote processor is set to remote role. 
This configuration allows to set the  CONFIG_OPENAMP_MASTER=n optimizing the OpenAMP library footprint for the remote processor which can have more memory constraint.
Another constraint is that the static vrings have to be initialized by the host before the remote can use it. If the remote processor is started by the host, this can lead to the use.

**Comment by carlocaione** [comment]
> Could you detail what you mean by "more close to the actual real-life usage"? 

What I'm thinking is a platform with 3 cores: A, B and C. They all want to communicate with each other. In this case it is likely that you will end up with instances on the same core that will be `host` with respect to one core but `remote` with respect to the other one.

```
[host] ─────instance───► [remote]

    ┌───┐                ┌───┐
    │ A ◄────────────────┤ B │
    └─┬─┘                └─▲─┘
      │                    │
      │                    │
      │                    │
      │                    │
      │                    │
      │        ┌───┐       │
      └────────► C ├───────┘
               └───┘
```
> The samples using OpenAMP have been designed to optimize the code for the remote side (i.e. the coprocessor). For this the main processor is usually set to host role and and the remote processor is set to remote role. This configuration allows to set the CONFIG_OPENAMP_MASTER=n optimizing the OpenAMP library footprint for the remote processor which can have more memory constraint.

Yes, but there could be cases in which a core can be `host` and `remote` at the same time on different instances. This change is showing that also this configuration (that, I agree, is unusual) is supported by the backend.

> Another constraint is that the static vrings have to be initialized by the host before the remote can use it. If the remote processor is started by the host, this can lead to the use.

But the backend (and the API) is meant to prevent that, indeed I can postpone the initialization of the `host` with respect to the `remote` and the sample will run just fine.

File: samples/subsys/ipc/ipc_service/static_vrings/remote/boards/nrf5340dk_nrf5340_cpunet.overlay [hunk 1]
```diff
@@ -49,7 +49,7 @@
 		memory-region = <&sram_ipc1>;
 		mboxes = <&mbox 2>, <&mbox 3>;
 		mbox-names = "rx", "tx";
-		role = "remote";
+		role = "host";
```
[the remark this answers is not in the document]
Comment (carlocaione) [reply samples/subsys/ipc/ipc_service/static_vrings/remote/boards/nrf5340dk_nrf5340_cpunet.overlay:52]: yeah, naming is a bit unfortunate. That should really be `remote_core`, `secondary` or something like that

**Comment by arnopo** [comment]
> > Could you detail what you mean by "more close to the actual real-life usage"?
> 
> What I'm thinking is a platform with 3 cores: A, B and C. They all want to communicate with each other. In this case it is likely that you will end up with instances on the same core that will be `host` with respect to one core but `remote` with respect to the other one.
> 
> ```
> [host] ─────instance───► [remote]
> 
>     ┌───┐                ┌───┐
>     │ A ◄────────────────┤ B │
>     └─┬─┘                └─▲─┘
>       │                    │
>       │                    │
>       │                    │
>       │                    │
>       │                    │
>       │        ┌───┐       │
>       └────────► C ├───────┘
>                └───┘

Thank for the clarification!
What about creating a specific sample for this configuration and keep existing one for host<->remote basic IPC?

**Comment by carlocaione** [comment]
> Thanks for the clarification!

Thanks for your reviews :)

> What about creating a specific sample for this configuration and keep existing one for host<->remote basic IPC?

Sure, no prob.

**Comment by carlescufi** [comment]
@carlocaione should we push this forward?

**Comment by carlocaione** [comment]
> @carlocaione should we push this forward?

Not actually needed anymore. Closing.

**Pull request closed without merging**